client/src/components/CardDetail.js

```diff
@@ -0,0 +1,493 @@
+        <FormControlLabel
+          control={
+            <Checkbox
+              checked={item.active}
+              onChange={handleChecklist}
+              name={index}
+              inputProps={{ 'aria-label': item.content }}
+            />
+          }
+          label={item.content}
+        />
+      ))}
+      <TextField
+        multiline={true}
+        rows={1}
+        type="text"
+        variant={"outlined"}
+        placeholder={"Add an item ..."}
+        size="small"
+        onKeyDown={handleKeyPress}
+      ></TextField>
+    </FormGroup>
+  );
+};
+
+const CardStyle = (theme) => ({
+  root: {
+    height: 700,
+    width: 840,
+  },
+  leftGrid: {
+    [theme.breakpoints.up('sm')] :{
+      width: '79%',
+    },
+    [theme.breakpoints.down('sm')] :{
+      width: '100%',
+    }
+  },
+  rightGrid: {
+    [theme.breakpoints.up('sm')] :{
+      width: '21%',
+    },
+    [theme.breakpoints.down('sm')] :{
+      width: '100%',
+    }
+  },
+  label: {
+    fontSize: 18,
+    fontWeight: "bold",
+    marginBottom: 8,
+  },
+  bottomButton: {
+    height: 34,
+    width: 100,
+    marginRight: theme.spacing(4),
+    marginTop: 20,
+  },
+  datePicker: {
+    textColor: "#759CFC",
+    color: "#759CFC",
+  },
+});
+
+const CardDetail = (props) => {
+  const { handleClose, open } = props;
+  const card = props.card;
+  const {cards, setCards, columns, setColumns} = useDashboard();
+
+  //States for showing the elements
+  const [showDeadline, setShowDeadline] = useState(undefined);
+  const [showTag, setShowTag] = useState(undefined);
+  const [showAttachment, setShowAttachment] = useState(undefined);
+  const [showChecklist, setShowChecklist] = useState(undefined);
+
+  const [deadline, setDeadline] = useState(card.deadline);
+  const [description, setDescription] = useState(card.description);
+  const [colorCode, setColorCode] = useState(card.colorCode);
+  const [tags, setTags] = useState(card.tags.join(', '));
+
+  const [isFetching, setIsFetching] = useState(false);
+  const [error, setError] = useState(undefined);
+
+  const [checklistItems, setChecklistItems] = useState(card.checklist);
+
+  const classes = makeStyles(CardStyle)();
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const url = '/api/v1/cards/'+card._id;
+    const content = {
+      deadline, description, colorCode, tags: tags.split(', '),
+      checklist: checklistItems,
+    }
+
+    setIsFetching(true);
+
+    authJSONFetch(url, {
+      method: 'PUT',
+      body: JSON.stringify(content)
+    }).then(res => res.json())
+      .then(data => {
+        setIsFetching(false);
+
+        if (data.error) {
+          throw new Error(data.error);
+          return;
+        }
+        setError(undefined);
+
+        //Update the card to Dashboard Provider
+        const newCards = {...cards};
+        newCards[card._id] = data;
+        setCards(newCards);
+
+        //Close the dialog
+        // handleClose();
```

Comment:
Can you uncomment this line and check if the dialog box closes on clicking save?